pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -278,3 +275,44 @@ func (c *Controller) getNodePortGatewayServices() []*model.Service {
 
 	return out
 }
+
+// gatewaySet is a helper to manage a set of NetworkGateway instances.
+type gatewaySet map[model.NetworkGateway]struct{}
+
+func (s gatewaySet) equals(other gatewaySet) bool {
+	return reflect.DeepEqual(s, other)
+}
+
+func (s gatewaySet) add(gw model.NetworkGateway) {
+	s[gw] = struct{}{}
+}
+
+func (s gatewaySet) addAll(other gatewaySet) {
+	for gw := range other {
+		s[gw] = struct{}{}
+	}
+}
+
+func (s gatewaySet) toArray() []*model.NetworkGateway {
+	gws := make([]*model.NetworkGateway, 0, len(s))
+	for gw := range s {
+		gw := gw
+		gws = append(gws, &gw)
+	}
+
+	// Sort the array so that it's stable.
+	sort.SliceStable(gws, func(i, j int) bool {
```

Comment:
Why do we need them sorted? It sounds like a familiar requirement somewhere but I forgot..

And if we do want them sorted, would it make more sense to sort in the place where we aggregate them?

Reply:
We probably don't actually need to sort. I just generally prefer to add stability to the ordering of the returned list.

WRT how we sort. I'm not sure it matters much ... there might be a slight performance improvement by sorting as we add, but there honestly won't be that many gateways to make a difference in practice. This code is simpler.